### ISSUE TITLE
javax.crypto.BadPaddingException: Decryption error while Decrypt the symmetric key

### ISSUE THREAD

**Issue by user370305** [microsoftgraph/microsoft-graph-docs#7346]
Below is code sample [Java]

    String rsaKeyAlgorithm = "RSA";
    String rsaEncryptAlgorithm = "RSA/ECB/OAEPWithSHA-256AndMGF1Padding";

    PrivateKey privateKey = (RSAPrivateKey) KeyFactory.getInstance(rsaKeyAlgorithm).generatePrivate(new PKCS8EncodedKeySpec(Base64.getDecoder().decode(privKey.getBytes("UTF-8"))));

    byte[] bytes = Base64.getDecoder().decode(dataKey.getBytes("UTF-8"));

    Cipher decriptCipher = Cipher.getInstance(rsaEncryptAlgorithm);
    OAEPParameterSpec oaepParams = new OAEPParameterSpec("SHA-256", "MGF1", new MGF1ParameterSpec("SHA-1"), PSource.PSpecified.DEFAULT);
    decriptCipher.init(Cipher.DECRYPT_MODE, privateKey, oaepParams);
    //decriptCipher.init(Cipher.DECRYPT_MODE, privateKey);
    System.out.println(new String(decriptCipher.doFinal(bytes)));

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 9e7bb652-e041-1ba5-9679-908a98e51a79
* Version Independent ID: 540d51d6-50d6-828c-2aa1-0bb02f30ee22
* Content: [Set up change notifications that include resource data (preview) - Microsoft Graph](https://docs.microsoft.com/en-us/graph/webhooks-with-resource-data#feedback)
* Content Source: [concepts/webhooks-with-resource-data.md](https://github.com/microsoftgraph/microsoft-graph-docs/blob/master/concepts/webhooks-with-resource-data.md)
* Product: **non-product-specific**
* Technology: **microsoft-graph**
* GitHub Login: @baywet
* Microsoft Alias: **MSGraphDocsVteam**

**Comment by baywet** [comment]
Hey @user370305 
Did you by any chance author this question as well?
https://stackoverflow.com/questions/60545312/how-to-decrypt-webhook-response-data-in-microsoft-graph
I'm working on a sample for Java to demonstrate end to end how webhooks with data should be implemented.

**Comment by user370305** [comment]
No. I came across that too but it is also unanswered. Thanks for noticing and acknowledging.

**Comment by baywet** [comment]
thanks for the additional details!
By any chance, could you upload a sanitized version of your project somewhere so I can take a look please?

**Comment by user370305** [comment]
Hey thanks for followup. I solved this, it was issue of RSA key generation.

**Comment by baywet** [comment]
@user370305 can you post the updated code here please? I'd like to add it to the sample I'm working on if you don't mind.

**Comment by user370305** [comment]
Sure. Just give me a time to sanitized it and I will share. Thanks again.

**Comment by baywet** [comment]
Nevermind I made good progress on the Java sample I'm building here (will move in the future) today. The only thing it's missing is token validation. https://github.com/baywet/msgraph-spring-webhooks-java-sample/blob/master/src/main/java/com/example/restservice/NotificationController.java
For reference your `rsaEncryptAlgorithm` should have been set to `RSA/ECB/OAEPWithSHA1AndMGF1Padding`.
Closing for now
